sklearn/cluster/k_means_.py

```diff
@@ -313,12 +320,25 @@ def k_means(X, n_clusters, init='k-means++', precompute_distances='auto',
     x_squared_norms = row_norms(X, squared=True)
 
     best_labels, best_inertia, best_centers = None, None, None
+    if n_clusters == 1:
```

Comment:
The mean will be selected after 1 iteration in this case, so what you are saying is done anyways. The only thing we can do is set `n_jobs` to `1` as no matter what the initialization, the algorithm would converge in 1 iteration